### ISSUE TITLE
Pressing "Save personal bindings" does not work for a built project.

### ISSUE THREAD

**Issue by peroht** [ValveSoftware/steamvr_unity_plugin#536]
SteamVR beta 1.8.5
Unity 2019.1.6f1.
SteamVR Unity Plugin v2.3.2.

Inside Unity, when opening the binding UI I am able to edit and save my personal bindings. When I build the project as a standalone build and try to edit then save my personal bindings the UI stalls. I have to restart SteamVR to be able to edit any application further.

![gamma one](https://user-images.githubusercontent.com/14336296/65460310-dfcd4600-de51-11e9-9c58-eacdebf21ca2.jpg)

**Comment by peroht** [comment]
I have found a temporary solution to this so for anyone interested this is how to bypass this bug for now.

Tested on SteamVR 1.7.15. 
Tested on various Unity and Unreal projects that are launched from the local harddrive (meaning they are not distributed on Steam).

When one tries to edit any legacy application, any changes made in the Devices>Controller Settings are immediately saved in a file located in my documents folder. The naming of the file is based on what controller type you choose. However, the naming has nothing to do with the application you are trying to edit. So, in my case, if I try to edit the controller settings for the vive controller for an app called "TestApp" the software will create the following file:

`C:\Users\pt\Documents\steamvr\input\_vive_controller.json`

If I edit the app for an index controller the software will create the following file:
`C:\Users\pt\Documents\steamvr\input\_knuckles.json`

If I edit the app for an omnideck controller the software will create the following file:
`C:\Users\pt\Documents\steamvr\input\_omnideck.json`

Looking in the same folder, one can see various locally saved bindings file with the naming scheme:
`C:\Users\pt\Documents\steamvr\input\steam.app.ANUMBER_ACONTROLLER.json`

This must clearly be a bug in SteamVR.

Now, for my case, starting with 1.7.15 - which introduced an ability to return the bindings  with lef/right hand in the extra settings - to make the app in question work with the Omnideck I copy the last options section from the file

`C:\Users\pt\Documents\steamvr\input\_omnideck.json`

```
   "options" : {
      "returnBindingsWithLeftHand" : true,
   }
```

and place it as the default in the legacy_bindings_omnideck.json file for my driver.

If this is already placed in the bindings file all is fine when you start a new application using the legacy input system. However, if you try to edit the legacy application in question you will end up with a stalling "Uploading..." notificiation in the UI and SteamVR will create the file in your local documents folder as stated above.

**Comment by CybershoesVR** [comment]
Yes, we've experienced the same issue on SteamVR 1.7 
For testing purposes, we bypassed it in a brutal way: Clicking "Replace default bindings" worked for us. Of course, it affects the other default bindings of your controller as well.

**Comment by peroht** [comment]
Happy to be of help,
keep hacking.

/P

**Comment by peroht** [comment]
This has been fixed by valve as of SteamVR beta 1.8.13.
